crates/wasi-common/cap-std-sync/src/net.rs

```diff
@@ -240,31 +170,14 @@ macro_rules! wasi_stream_write_impl {
             fn as_any(&self) -> &dyn Any {
                 self
             }
-            async fn sock_accept(&mut self, _fdflags: FdFlags) -> Result<Box<dyn WasiFile>, Error> {
-                Err(Error::badf())
-            }
-            async fn datasync(&self) -> Result<(), Error> {
-                Err(Error::badf())
-            }
-            async fn sync(&self) -> Result<(), Error> {
-                Err(Error::badf())
-            }
-            async fn get_filetype(&self) -> Result<FileType, Error> {
+            async fn get_filetype(&mut self) -> Result<FileType, Error> {
                 Ok(FileType::SocketStream)
             }
             #[cfg(unix)]
-            async fn get_fdflags(&self) -> Result<FdFlags, Error> {
+            async fn get_fdflags(&mut self) -> Result<FdFlags, Error> {
                 let fdflags = self.0.as_filelike().get_fd_flags()?;
                 Ok(from_sysif_fdflags(fdflags))
             }
-            #[cfg(windows)]
-            async fn get_fdflags(&self) -> Result<FdFlags, Error> {
-                // There does not seem to be a way for windows to call s.th. like `fcntl(fd, F_GETFL, 0)`
-                // on a socket.
-                // `rustix::fs::fcntl` is only available for Unix.
-                // `rustix::io::ioctl_fionbio` only sets the flags, but does not read.
-                Ok(FdFlags::empty())
-            }
```

Comment:
I think the intention here is that it will eventually be implemented once it becomes possible.